aws/request.go

```diff
@@ -111,12 +112,8 @@ func New(cfg Config, metadata Metadata, handlers Handlers,
 
 	if err == nil {
 		// TODO so ugly
-		metadata.Endpoint = endpoint.URL
-		if len(endpoint.SigningName) > 0 && !endpoint.SigningNameDerived {
-			metadata.SigningName = endpoint.SigningName
-		}
-		if len(endpoint.SigningRegion) > 0 {
-			metadata.SigningRegion = endpoint.SigningRegion
```

Comment:
Looking at this again, with this change, it should probably have a similar check as the new code I wrote below which checks to see if the returned endpoint had an empty signing region, and fallback to the metadata signing region. Metadata signing region would be initialized with the aws.Config.Region as part of client construction.